src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -2644,12 +2640,18 @@ void MFEMSidreDataCollection::reconstructField(Group* field_grp)
     else
     {
       m_owned_quadfuncs.emplace_back(
-        new mfem::QuadratureFunction(m_quadspaces.at(basis_name).get(),
-                                     values,
-                                     vdim));
+        new mfem::QuadratureFunction(m_quadspaces.at(basis_name).get(), vdim));
+
+      // FIXME: A bug was fixed in QuadratureFunction::SetSpace after mfem@4.5
+      // See: https://github.com/mfem/mfem/pull/3281
+      // Update using the following when possible to avoid an unnecessary allocation in the constructor
+      // (1) Use default Quadrature function constructor
+      // (2) Use qf->SetSpace(qs, values, vdim);
+      auto* qf = m_owned_quadfuncs.back().get();
+      qf->NewDataAndSize(values, vdim * qf->GetSpace()->GetSize());
```

Comment:
Note: mfem refactored the `QuadratureFunction` API between mfem@4.4 and mfem@4.5, removing the constructor that allowed directly passing in the data. Presumably, this was not intentional. I will create a PR in mfem to add it back so we can go back to the simpler form for the next release of mfem.

There was also a bug in the mfem@4.5 release that does not allow us to use the `QuadratureFunction::SetSpace` function. It was fixed in https://github.com/mfem/mfem/pull/3281, but that was merged after the mfem@4.5 release.

As a consequence, it looks like we'll have an unnecessary allocation for the underlying vector in the meantime.

Tag: @tzanio @v-dobrev @rcarson3 @jamiebramwell

Reply:
This is unfortunate omission in MFEM v4.5.

If you want to use an MFEM release version with this issue fixed, maybe we can create a bugfix release, v4.5.2, after bringing back the constructor.

Reply:
Thanks for the offer @v-dobrev! We'd be happy to upgrade axom's mfem once https://github.com/mfem/mfem/pull/3362 gets merged. 

As far as I can tell, the current workaround will require an unnecessary memory allocation for the data associated with each `QuadratureField` during a restart. I think this affects @jamiebramwell's code more than mine, but I wouldn't expect it to noticeably affect the overall runtime/performance.

Reply:
I need to rework our quadrature function storage to work on arbitrary types anyway, so this shouldn't affect us.

Reply:
@kennyweiss, when you refer to _"axom's mfem"_, how is this mfem version (or commit) set?

Reply:
Looking at this PR, it seems that the required version of MFEM is precisely the v4.5 release and that seems to be hard-coded in several places. So with this PR, the only acceptable MFEM version becomes v4.5, right?

Reply:
We're currently only maintaining support for a single version of mfem, which we set in our spack config and list in our `RELEASE-NOTES` (see the diffs in this PR).  

If we needed to, I suppose we could add support for different/multiple mfem versions via `#ifdef`s.

Reply:
Okay, I think I understand -- in order to get bugfixes in MFEM available for use in Axom, we need to make them part of a release.

Let us discuss a possible MFEM v4.5.2 release and we'll let you know when it's ready to be tested with Axom before we make the release.